SIPS/sip-21.md

```diff
@@ -0,0 +1,53 @@
+---
+sip: 21
+title: Snap-defined timeouts
+status: Draft
+discussions-to: https://github.com/MetaMask/SIPs/discussions/128
+author: Frederik Bolding (@frederikbolding)
+created: 2024-01-12
+---
+
+## Abstract
+
+This SIP describes a way for Snaps to customize the request timeouts that constrain the Snap lifecycle. Effectively allowing the Snap to opt-in to slightly longer or shorter timeouts where needed.
```

Comment:
What's the use case for a Snap requesting shorter timeouts?

Reply:
If a snap dev is confident that their snap runs faster than the 1 minute timeout and therefore confident that if it takes longer than 15 seconds something is wrong I think it makes sense for them to be able to use that as the timeout.

If something is actually wrong the user would be notified much quicker, making the UX slightly better.